### PR TITLE
Move ResourceTree to React

### PR DESCRIPTION
**Pull request by languy**
* Move `ResourceTreeAdapter` code to `useNotebooks` hook.
* Remove unnecessary `triggerRender()` calls

### CODE REVIEW

File: src/Main.tsx [hunk 1]
```diff
@@ -158,7 +186,15 @@ const App: React.FunctionComponent = () => {
...
-                  <div style={{ overflowY: "auto" }} data-bind="if: !isAuthWithResourceToken(), react:resourceTree" />
+                  <div style={{ overflowY: "auto" }} data-bind="if: !isAuthWithResourceToken()">
+                    <ResourceTree
+                      explorer={explorer}
+                      lastRefreshedTime={lastRefreshTime}
+                      galleryContentRoot={galleryContentRoot}
+                      myNotebooksContentRoot={myNotebooksContentRoot}
+                      gitHubNotebooksContentRoot={gitHubNotebooksContentRoot}
+                    />
+                  </div>
```
Comment (notlaforge) [inline src/Main.tsx:197]:
```suggestion
                  <div style={{ overflowY: "auto" }}>
                    isAuthWithResourceToken() || <ResourceTree
                      explorer={explorer}
                      lastRefreshedTime={lastRefreshTime}
                      galleryContentRoot={galleryContentRoot}
                      myNotebooksContentRoot={myNotebooksContentRoot}
                      gitHubNotebooksContentRoot={gitHubNotebooksContentRoot}
                    />
                  </div>
```

Comment (notlaforge) [reply src/Main.tsx:197]: Is this an opportunity to remove one more knockout binding?

File: src/Main.tsx [hunk 2]
```diff
@@ -99,10 +113,24 @@ const App: React.FunctionComponent = () => {
...
     closeDialog,
+    onRefreshNotebookList: refreshList,
+    initializeGitHubRepos,
+    getMyNotebooksContentRoot,
   };
   const config = useConfig();
   const explorer = useKnockoutExplorer(config?.platform, explorerParams);
 
+  // TODO fix this
+  context.container = explorer;
```
Comment (southpolesteve) [inline src/Main.tsx:124]: I have some ideas about how to avoid this

File: src/Main.tsx (hunk 1 above)
Comment (languy) [reply src/Main.tsx:188]: It's a bit more involved as `isAuthWithResourceToken` isn't available in `Main.tsx`, but in `Explorer` (btw, you probably meant to enclose this expression inside `{{ }}`). So we'll first move everything to react, then will come back to remove the ko's (ie replace them as some state in `Main.tsx`).

**Pull request closed without merging**